### PR TITLE
Parser changes for Gecko integration

### PR DESCRIPTION
**Pull request by jdm**
This is a grab bag of changes that were important for getting Stylo tests passing.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-cssparser/168)

### CODE REVIEW

File: src/tokenizer.rs [hunk 1]
```diff
@@ -226,7 +226,7 @@ impl<'a> Tokenizer<'a> {
             input: input,
             position: 0,
             last_known_source_location: Cell::new((SourcePosition(0),
-                                                   SourceLocation { line: 1, column: 1 })),
+                                                   SourceLocation { line: 0, column: 0 })),
```
Comment (SimonSapin) [inline src/tokenizer.rs]: `SourceLocation`’s doc-comments needs to up updated accordingly.

File: src/parser.rs [hunk 2]
```diff
@@ -188,6 +188,11 @@ impl<'i: 't, 't> Parser<'i, 't> {
         }
     }
 
+    /// Return the current line that is being parsed.
```
Comment (SimonSapin) [inline src/parser.rs]: Newlines are not significant in CSS. Why is the current line relevant? Is this counting on authors not using minification, adding newlines between declarations, and not writing multi-line declarations?

Comment (jdm) [inline src/parser.rs]: Gecko's error messages include a `sourceLine` property which is expected to be the current line being parsed.

**Comment by jdm** [comment]
Related Gecko work is in the second patch of https://bugzilla.mozilla.org/show_bug.cgi?id=1352669.

**Comment by SimonSapin** [comment]
r+ with the doc-comment updated for line and columns numbers starting at 0 rather than 1.

I still think that "current line" is not the right thing to do, but I suppose adding it will help enable these tests and get more test coverage of other things more easily than changing the tests to do something else. So let’s add it.

**Comment by SimonSapin** [comment]
@jdm It looks like you landed this without the doc-comment change I requested. Could you fix that? CC https://bugzilla.mozilla.org/show_bug.cgi?id=1380890

**Comment by jdm** [comment]
Weird; I remember making that change.

**Comment by jdm** [comment]
Wait, I did: https://github.com/servo/rust-cssparser/pull/168/files#diff-0bd70f3585cde6d2d2aff02435db9070L389

**Comment by SimonSapin** [comment]
I misread the diff, sorry.

**Pull request merged**